### DISCUSSION TITLE
Constrained variables are, apparently, non-functional.

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [PowerShell/PowerShell#20885] [Q&A] [answered]
I'm trying to bind variables to a specific type[^0] using https://learn.microsoft.com/en-us/powershell/scripting/lang-spec/chapter-05?view=powershell-5.1#53-constrained-variables, but this doesn't appear to function:

```.log
PowerShell Extension v2023.11.1
```

1.	```.PS1
	[Int32]$ExampleVariable
	```

	outputs

	> ```.log
	> 0
	> ```

1.	```.PS1
	$ExampleVariable = 1   
	$ExampleVariable
	```

	outputs

	> ```.log
	> 1
	> ```

1.	```.PS1
	$ExampleVariable = 'a'        
	$ExampleVariable
	```

	outputs

	> ```.log
	> a
	> ```

[^0]: https://stackoverflow.com/a/1517670/9731176)

**Comment by SeeminglyScience** [comment, marked as answer]
You need an assignment, e.g. `[int] $ExampleVariable = 0`. Otherwise it's just a normal explicit conversion (like `[int]'23'` returns `23` as an `int`)

**Reply by roke-julian-lockhart** [reply]
> https://github.com/orgs/PowerShell/discussions/20885#discussioncomment-7812633

Thank you, @SeeminglyScience:

<blockQuote>

```.log
PS /home/RokeJulianLockhart> [int] $ExampleVariable2 = 0
PS /home/RokeJulianLockhart> $ExampleVariable2          
0
PS /home/RokeJulianLockhart> $ExampleVariable2 = 'a'    
MetadataError: Cannot convert value "a" to type "System.Int32". Error: "The input string 'a' was not in a correct format."
PS /home/RokeJulianLockhart>
```

</blockQuote>
